src/solvers/README.md

```diff
@@ -62,7 +62,111 @@ different decision procedures, roughly one per directory.
 
 \section flattening-section Flattening
 
-To be documented.
+**Key classes:**
+* \ref propt
+* \ref boolbvt
+* \ref bv_utilst
+
+The main class in this folder is \ref boolbvt which wraps a variety of 
+helper methods and classes (including inherited ones) and acts as an interface 
+for transforming \ref exprt into boolean formula and then solving it.
+
+Many of its methods are focused around transforming a particular \ref exprt 
+into a vector of \ref literalt and then passing them to a \ref propt 
+for formula building / solving.
+
+The primary methods are:
+
+`bvt boolbvt::convert_bitvector(const exprt &expr)`
+
+Which takes an exprt then calls the associated transformation functions to
+generate the \ref literalt vector to then pass to the internal \ref propt instance.
+
+`const bvt & boolbvt::convert_bv(const exprt &expr, optionalt<std::size_t> expected_width)`
+
+Similar to convert_bitvector except it also provides basic caching and
+freezing results for incremental generation. It calls convert_bitvector 
```

Comment:
I don't know what `incremental generation` is. Is that described somewhere?

Reply:
Incremental solving by the SAT solver, there are a few references to it: 

> Freezing variables is necessary to make use of incremental solving with MiniSat SimpSolver. Potentially a useful hook for other applications using incremental solving.

I didn't look into precisely what that entails though.

Reply:
`incremental solving` is a well-known concept. But I've never heard of `incremental generation`...

Reply:
I just wrote it as shorthand for incremental generation of the formula, but I'll tighten it up.